### PR TITLE
Preserve enum value case when emitting type metadata

### PR DESCRIPTION
**Pull request by fulghum**
Depends on: https://github.com/dolthub/go-mysql-server/pull/1179

### CODE REVIEW

File: integration-tests/bats/migration-integration.bats [hunk 1]
```diff
@@ -21,15 +21,15 @@ teardown() {
     run dolt tag -v
     [ "$status" -eq 0 ]
     [[ "$output" =~ "r9jv07tf9un3fm1fg72v7ad9er89oeo7" ]] || false
-    [[ ! "$output" =~ "tetumstdaa81t25h8p5tstsb1f94r4q1" ]] || false
+    [[ ! "$output" =~ "vkoodfe2im3pei1o0vurk66lna6jqhlc" ]] || false
```
Comment (fulghum) [inline integration-tests/bats/migration-integration.bats:24]: I was surprised these tests broke. If there's a case-sensitive schema that's used as part of creating the commit hash, that would make sense about why this changed though.

Comment (zachmu) [reply integration-tests/bats/migration-integration.bats:24]: This is why it's happening:

Here's what we write to disk when serializing schema:

```
	for i := len(cols) - 1; i >= 0; i-- {
		col := cols[i]
...
		to := b.CreateString(sqlTypeString(col.TypeInfo))
...
		serial.ColumnAddSqlType(b, to)
	}
```

That method does this:

```
func sqlTypeString(t typeinfo.TypeInfo) string {
	typ := t.ToSqlType()
	if st, ok := typ.(sql.SpatialColumnType); ok {
		// for spatial types, we must append the SRID
		if srid, ok := st.GetSpatialTypeSRID(); ok {
			return fmt.Sprintf("%s SRID %d", typ.String(), srid)
		}
	}
	return typ.String()
}```

So when you changed the `String()` methods of all those types in GMS, it actually changed the commit hashes of anything using those databases.

Do we care about this enough to bump the feature version?

**Review by zachmu — APPROVED** [review]
I think this is fine but let's ask the team about the commit hash changing

File: integration-tests/bats/migration-integration.bats (hunk 1 above)
Comment (fulghum) [reply integration-tests/bats/migration-integration.bats:24]: Just closing the loop here after the team discussion... I followed up with additional testing for diff and merge scenarios, particularly looking to make sure Dolt didn't interpret the difference in schema hashes (with the new casing) as a schema change. After testing with the old and new Dolt versions, everything was looking good for diff and merge so we're moving ahead.

**Pull request merged**